src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String line;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = bufferedReader.readLine()) != null) {
+                String actionType = line.split(COMMA)[0];
+                int amount = Integer.parseInt(line.split(COMMA)[1]);
+
+                if (actionType.equals(SUPPLY)) {
+                    supplyAmount += amount;
+                }
+
+                if (actionType.equals(BUY)) {
+                    buyAmount += amount;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Cannot find file", e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        writeResult(supplyAmount, buyAmount, toFileName);
```

Comment:
1. Generally it's not a good practice to perform business logic during keeping active connection to file.
Please make reading of full content first then work with content

2. As you will have 3 logical blocks (read file, generate report, write report top file) please make separate private method for each of these blocks and only call them in public method in order to make it short and simple